plugins/modules/content_template.py

```diff
@@ -206,7 +206,7 @@ def main():
     argument_spec.update(
         template=dict(type='str', required=True),
         library=dict(type='str', required=True),
-        vm_name=dict(type='str', required=True),
+        vm_name=dict(type='str', required=(argument_spec['state'] == 'present')),
```

Comment:
this is not the correct syntax, see `required-if` here 
https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec-dependencies